src/loaders/workspace_downloader/workspace_downloader.py

```diff
@@ -280,29 +295,119 @@ def process_input(conf):
-        # upa file is downloaded to cfn
-        conf.asu.get_assembly_as_fasta({"ref": upa.replace("_", "/"), "filename": upa})
+    # retrieve data links associated with upa
+    links_ret = conf.ss.get_data_links_from_data({"upa": upa.replace("_", "/")})
 
-        # each upa in output_dir as a seperate directory
-        dstd = os.path.join(conf.pth, upa)
-        os.makedirs(dstd, exist_ok=True)
+    data_links = links_ret['links']
+    if not data_links and conf.ignore_no_sample_error:
+        print(f"No sample data links found for {upa}")
+        return
 
-        dst = os.path.join(dstd, f"{upa}.fa")
-        # Hard link .fa file from job_dir to output_dir in WS
-        os.link(cfn, dst)
+    # there should only be one data link for each upa
+    if len(data_links) != 1:
+        raise ValueError(f"Expected 1 data link for {upa}, got {len(data_links)}")
 
-        metafile = os.path.join(dstd, f"{upa}.meta")
-        # save meta file with relevant object_info
-        with open(metafile, "w", encoding="utf8") as json_file:
-            json.dump(_process_object_info(obj_info, genome_upa), json_file, indent=2)
+    # retrieve sample data and save to file
+    sample_id = data_links[0]['id']
+    sample_ret = conf.ss.get_sample_via_data({"upa": upa.replace("_", "/"),
+                                              "id": sample_id,
+                                              "version": data_links[0]["version"]})
 
-        print("Completed %s" % (upa))
+    with open(sample_file, "w", encoding="utf8") as json_file:
+        json.dump(sample_ret, json_file, indent=2)
+
+    meta[loader_common_names.SAMPLE_FILE_KEY] = sample_file_name
+    with open(metafile, "w", encoding="utf8") as json_file:
+        json.dump(meta, json_file, indent=2)
+
+    sample_node_name = f"{upa}.{loader_common_names.SAMPLE_NODE_FILE_EXT}"
+    sample_node_file = os.path.join(upa_dir, sample_node_name)
+    if os.path.isfile(sample_node_file):
+        print(f"Skip generating sample node tree for {upa} as it already exists")
+        return
+
+    try:
+        node_data = _retrieve_node_data(sample_ret['node_tree'])
+    except BadNodeTreeError as e:
+        raise ValueError(f"Error retrieving node data for {upa}") from e
+
+    with open(sample_node_file, "w", encoding="utf8") as json_file:
+        json.dump(node_data, json_file, indent=2)
+
+
+def _retrieve_node_data(
+        node_tree: list[dict[str, Any]]
+) -> dict[str, str | int | float]:
+    # retrieve the meta_controlled node data from node tree
+
+    node_data = dict()
+
+    if len(node_tree) != 1:
+        raise BadNodeTreeError(f"Expected 1 node in node tree, got {len(node_tree)}")
+
+    sample_node = node_tree[0]
+
+    meta_controlled = sample_node['meta_controlled']
+    for key, meta_value in meta_controlled.items():
+
+        if 'value' not in meta_value:
+            raise BadNodeTreeError(f"Expected 'value' key in meta_value, got {meta_value}")
+
```

Comment:
A couple other things:
* We should throw an error if there's anything other than "value" (or also units for the lat and long keys, in which case we should check the units are degrees). Otherwise we may be silently ignoring data.
* For the lat and long keys:
    * We need to ensure that all samples that have lat and long data have that data in the same keys
        * Non-GROW samples may have different key names for lat and long, we may have to expand this code later to change the key names
    * The samples data product will need to add indexes on those keys so the key names should be in the common names file

Reply:
Actually, given :https://www.arangodb.com/docs/stable/indexing-geo.html

We should probably put the lat and long into an array.

Reply:
Oy... for now let's do both. We can hide the array from users but we'll have it for geo queries if needed

Reply:
:+1: